### DISCUSSION TITLE
Not able to include analytical corrections for SCPH bands calculations

### DISCUSSION THREAD

**Discussion by eeshaAnd** [ttadano/alamode#86] [Q&A]
Hi,
  I have tried different values of hyper-parameter alpha ranging from 0.1 to 0.01 and convergence threshold from 1E-5 to 1E-9. But, when I include born effective charges, the results do not converge for all the temperatures. If they converge for 300K, it gives an anomaly in the graph ( a sudden jump in the phonon branches). How to address this problem? What input parameters do I need to change?

Best,
Eesha

**Comment by ttadano** [comment]
Hi, please tell me more detailed information about the computational conditions including

- supercell size for harmonic and anharmonic IFC calculations
- `KMESH_SCPH` and `KMESH_INTERPOLATE`
- `NONANALYTIC` option (1 or 2)

**Reply by eeshaAnd** [reply]
Here are the input parameters for my calculations:
&general
  PREFIX = ins_551_scph
  MODE = SCPH
  FCSXML = ins_551_quartic_opt.xml
  NKD = 2
  KD = In S
  NONANALYTIC = 2
  BORNINFO = ins.born
  TMIN = 0; TMAX = 500
  DT = 50
/

&scph
 SELF_OFFDIAG = 0
 MAXITER = 20000
 MIXALPHA = 0.01
 KMESH_INTERPOLATE = 5 5 1
 KMESH_SCPH = 10 10 1
 TOL_SCPH = 5.0e-8
/

&cell
60.0000  # factor in Bohr unit
0.1238893368 0.0110727264 0.0000000000
0.0523220999 0.1128266533 0.0000000000
0.0000000000 0.0000000000 0.9724542255
/

&kpoint
  1  # KPMODE = 1: line mode
  G 0.0 0.0 0.0   M 0.5 0.0 0.0 51
  M 0.5 0.0 0.0   K 0.67 0.33 0.0 51
  K 0.67 0.33 0.0 G 0.0 0.0 0.0 51
/

**Comment by eeshaAnd** [comment]
Dear Dr. Tadano,
     I have been unable to solve the following two problems regarding my calculation son monolayer InS.

  1.  Discussion 131 (https://github.com/ttadano/alamode/discussions/131) - How to solve the memory issue while using alm calculator in Harmonic phonon calculations?
  2.  We find an oscillating behavior in the lattice cubic thermal conductivity while checking convergence w.r.t. q grid. PFA the graph. Also, the value of Kl is not same as that of literature. I know we use Finite displacement method and fit it to OLS and then solve BTE within RTA. Does this differ from codes like Phonopy and ShengBTE? If so, how? What could give rise to such an oscillating behavior in convergence?

Please help!

Best regards,
Eesha

  1.


________________________________

**Reply by ttadano** [reply]
Hi,

About the memory issue, please have a look at https://github.com/ttadano/alamode/discussions/47#discussioncomment-1433991.

The kL value tends to be slighly larger when an odd number is used for the q grid. So, increasing the number of q point sampling one-by-one can lead to an oscillatory behavior.

**Comment by eeshaAnd** [comment]
Okay! Thank you, Dr. Tadano!
But, this value is not similar to that of literature although it has been obtained by solving BTE equation (Literature values using ShengBTE is 0.4 and 0.55). But the third order Kl I obtain is 0.9 W/mK.
If I want to explain or highlight how our alculations may be different then literature, what can I state in the paper? Any pointers?

Best regards,
Eesha
________________________________

**Reply by ttadano** [reply]
When the same computational parameters and force constants are used, both codes should give very similar results at least within the RTA (3ph) level. 

Please examine if your computational settings and those of the mentioned previous work are consistent with each other. For 2D, you will also need to use the same value of the layer thickness.

**Comment by eeshaAnd** [comment]
Dear Dr. Tadano,
   Our layer thickness is higher than that of literature. Thank you for the prompt response.

Best regards,
Eesha
________________________________

**Comment by eeshaAnd** [comment]
Dear Dr. Tadano,
     I am still facing memory issues while running alm to produce log2 file to estimate the force constants using 10000MB memory. It reaches OLS step and stops.
Is it because of the alamode version? 
Here is my alm.in file:

&general
  PREFIX = ins_551
  MODE = optimize
  NAT = 144; NKD = 2
  KD = In S
/

&interaction
  NORDER = 1  # 1: harmonic, 2: cubic, ..
/

&optimize
  DFSET = DFSET_harmonic
  SPARSE = 1
/
&fitting
DFSET = DFSET_harmonic
/

&cell
60.0000  # factor in Bohr unit
0.7433360206345580	0.0664363582717853	0.0000000000000000 #a1
0.3139325991203610	0.6769599199272620	0.0000000000000000 #a2
0.0000000000000000	0.0000000000000000	0.9724542254726610 #a3
/

&cutoff 
  In-In None
  S-S None
  In-S  None
/

&position
1	0.0677094810408245	0.0494148820550885	0.0459060696180882
1	0.2343761477074910	0.0494148820550885	0.0459060696180882
1	0.4010428143741570	0.0494148820550885	0.0459060696180882
1	0.5677094810408240	0.0494148820550885	0.0459060696180882
1	0.7343761477074910	0.0494148820550885	0.0459060696180882
1	0.9010428143741570	0.0494148820550885	0.0459060696180882
1	0.0677094810408245	0.2160815487217550	0.0459060696180882
1	0.2343761477074910	0.2160815487217550	0.0459060696180882
1	0.4010428143741570	0.2160815487217550	0.0459060696180882
1	0.5677094810408240	0.2160815487217550	0.0459060696180882
1	0.7343761477074910	0.2160815487217550	0.0459060696180882
1	0.9010428143741570	0.2160815487217550	0.0459060696180882
1	0.0677094810408245	0.3827482153884210	0.0459060696180882
1	0.2343761477074910	0.3827482153884210	0.0459060696180882
1	0.4010428143741570	0.3827482153884210	0.0459060696180882
1	0.5677094810408240	0.3827482153884210	0.0459060696180882
1	0.7343761477074910	0.3827482153884210	0.0459060696180882
1	0.9010428143741570	0.3827482153884210	0.0459060696180882
1	0.0677094810408245	0.5494148820550880	0.0459060696180882
1	0.2343761477074910	0.5494148820550880	0.0459060696180882
1	0.4010428143741570	0.5494148820550880	0.0459060696180882
1	0.5677094810408240	0.5494148820550880	0.0459060696180882
1	0.7343761477074910	0.5494148820550880	0.0459060696180882
1	0.9010428143741570	0.5494148820550880	0.0459060696180882
1	0.0677094810408245	0.7160815487217550	0.0459060696180882
1	0.2343761477074910	0.7160815487217550	0.0459060696180882
1	0.4010428143741570	0.7160815487217550	0.0459060696180882
1	0.5677094810408240	0.7160815487217550	0.0459060696180882
1	0.7343761477074910	0.7160815487217550	0.0459060696180882
1	0.9010428143741570	0.7160815487217550	0.0459060696180882
1	0.0677094810408245	0.8827482153884210	0.0459060696180882
1	0.2343761477074910	0.8827482153884210	0.0459060696180882
1	0.4010428143741570	0.8827482153884210	0.0459060696180882
1	0.5677094810408240	0.8827482153884210	0.0459060696180882
1	0.7343761477074910	0.8827482153884210	0.0459060696180882
1	0.9010428143741570	0.8827482153884210	0.0459060696180882
1	0.0677094810408245	0.0494148820550885	0.9540939303819110
1	0.2343761477074910	0.0494148820550885	0.9540939303819110
1	0.4010428143741570	0.0494148820550885	0.9540939303819110
1	0.5677094810408240	0.0494148820550885	0.9540939303819110
1	0.7343761477074910	0.0494148820550885	0.9540939303819110
1	0.9010428143741570	0.0494148820550885	0.9540939303819110
1	0.0677094810408245	0.2160815487217550	0.9540939303819110
1	0.2343761477074910	0.2160815487217550	0.9540939303819110
1	0.4010428143741570	0.2160815487217550	0.9540939303819110
1	0.5677094810408240	0.2160815487217550	0.9540939303819110
1	0.7343761477074910	0.2160815487217550	0.9540939303819110
1	0.9010428143741570	0.2160815487217550	0.9540939303819110
1	0.0677094810408245	0.3827482153884210	0.9540939303819110
1	0.2343761477074910	0.3827482153884210	0.9540939303819110
1	0.4010428143741570	0.3827482153884210	0.9540939303819110
1	0.5677094810408240	0.3827482153884210	0.9540939303819110
1	0.7343761477074910	0.3827482153884210	0.9540939303819110
1	0.9010428143741570	0.3827482153884210	0.9540939303819110
1	0.0677094810408245	0.5494148820550880	0.9540939303819110
1	0.2343761477074910	0.5494148820550880	0.9540939303819110
1	0.4010428143741570	0.5494148820550880	0.9540939303819110
1	0.5677094810408240	0.5494148820550880	0.9540939303819110
1	0.7343761477074910	0.5494148820550880	0.9540939303819110
1	0.9010428143741570	0.5494148820550880	0.9540939303819110
1	0.0677094810408245	0.7160815487217550	0.9540939303819110
1	0.2343761477074910	0.7160815487217550	0.9540939303819110
1	0.4010428143741570	0.7160815487217550	0.9540939303819110
1	0.5677094810408240	0.7160815487217550	0.9540939303819110
1	0.7343761477074910	0.7160815487217550	0.9540939303819110
1	0.9010428143741570	0.7160815487217550	0.9540939303819110
1	0.0677094810408245	0.8827482153884210	0.9540939303819110
1	0.2343761477074910	0.8827482153884210	0.9540939303819110
1	0.4010428143741570	0.8827482153884210	0.9540939303819110
1	0.5677094810408240	0.8827482153884210	0.9540939303819110
1	0.7343761477074910	0.8827482153884210	0.9540939303819110
1	0.9010428143741570	0.8827482153884210	0.9540939303819110
1	0.1232748134082820	0.1049710828571930	0.0849083664066242
2	0.2899414800749480	0.1049710828571930	0.0849083664066242
2	0.4566081467416150	0.1049710828571930	0.0849083664066242
2	0.6232748134082810	0.1049710828571930	0.0849083664066242
2	0.7899414800749480	0.1049710828571930	0.0849083664066242
2	0.9566081467416150	0.1049710828571930	0.0849083664066242
2	0.1232748134082820	0.2716377495238590	0.0849083664066242
2	0.2899414800749480	0.2716377495238590	0.0849083664066242
2	0.4566081467416150	0.2716377495238590	0.0849083664066242
2	0.6232748134082810	0.2716377495238590	0.0849083664066242
2	0.7899414800749480	0.2716377495238590	0.0849083664066242
2	0.9566081467416150	0.2716377495238590	0.0849083664066242
2	0.1232748134082820	0.4383044161905260	0.0849083664066242
2	0.2899414800749480	0.4383044161905260	0.0849083664066242
2	0.4566081467416150	0.4383044161905260	0.0849083664066242
2	0.6232748134082810	0.4383044161905260	0.0849083664066242
2	0.7899414800749480	0.4383044161905260	0.0849083664066242
2	0.9566081467416150	0.4383044161905260	0.0849083664066242
2	0.1232748134082820	0.6049710828571930	0.0849083664066242
2	0.2899414800749480	0.6049710828571930	0.0849083664066242
2	0.4566081467416150	0.6049710828571930	0.0849083664066242
2	0.6232748134082810	0.6049710828571930	0.0849083664066242
2	0.7899414800749480	0.6049710828571930	0.0849083664066242
2	0.9566081467416150	0.6049710828571930	0.0849083664066242
2	0.1232748134082820	0.7716377495238590	0.0849083664066242
2	0.2899414800749480	0.7716377495238590	0.0849083664066242
2	0.4566081467416150	0.7716377495238590	0.0849083664066242
2	0.6232748134082810	0.7716377495238590	0.0849083664066242
2	0.7899414800749480	0.7716377495238590	0.0849083664066242
2	0.9566081467416150	0.7716377495238590	0.0849083664066242
2	0.1232748134082820	0.9383044161905260	0.0849083664066242
2	0.2899414800749480	0.9383044161905260	0.0849083664066242
2	0.4566081467416150	0.9383044161905260	0.0849083664066242
2	0.6232748134082810	0.9383044161905260	0.0849083664066242
2	0.7899414800749480	0.9383044161905260	0.0849083664066242
2	0.9566081467416150	0.9383044161905260	0.0849083664066242
2	0.1232748134082820	0.1049710828571930	0.9150916335933750
2	0.2899414800749480	0.1049710828571930	0.9150916335933750
2	0.4566081467416150	0.1049710828571930	0.9150916335933750
2	0.6232748134082810	0.1049710828571930	0.9150916335933750
2	0.7899414800749480	0.1049710828571930	0.9150916335933750
2	0.9566081467416150	0.1049710828571930	0.9150916335933750
2	0.1232748134082820	0.2716377495238590	0.9150916335933750
2	0.2899414800749480	0.2716377495238590	0.9150916335933750
2	0.4566081467416150	0.2716377495238590	0.9150916335933750
2	0.6232748134082810	0.2716377495238590	0.9150916335933750
2	0.7899414800749480	0.2716377495238590	0.9150916335933750
2	0.9566081467416150	0.2716377495238590	0.9150916335933750
2	0.1232748134082820	0.4383044161905260	0.9150916335933750
2	0.2899414800749480	0.4383044161905260	0.9150916335933750
2	0.4566081467416150	0.4383044161905260	0.9150916335933750
2	0.6232748134082810	0.4383044161905260	0.9150916335933750
2	0.7899414800749480	0.4383044161905260	0.9150916335933750
2	0.9566081467416150	0.4383044161905260	0.9150916335933750
2	0.1232748134082820	0.6049710828571930	0.9150916335933750
2	0.2899414800749480	0.6049710828571930	0.9150916335933750
2	0.4566081467416150	0.6049710828571930	0.9150916335933750
2	0.6232748134082810	0.6049710828571930	0.9150916335933750
2	0.7899414800749480	0.6049710828571930	0.9150916335933750
2	0.9566081467416150	0.6049710828571930	0.9150916335933750
2	0.1232748134082820	0.7716377495238590	0.9150916335933750
2	0.2899414800749480	0.7716377495238590	0.9150916335933750
2	0.4566081467416150	0.7716377495238590	0.9150916335933750
2	0.6232748134082810	0.7716377495238590	0.9150916335933750
2	0.7899414800749480	0.7716377495238590	0.9150916335933750
2	0.9566081467416150	0.7716377495238590	0.9150916335933750
2	0.1232748134082820	0.9383044161905260	0.9150916335933750
2	0.2899414800749480	0.9383044161905260	0.9150916335933750
2	0.4566081467416150	0.9383044161905260	0.9150916335933750
2	0.6232748134082810	0.9383044161905260	0.9150916335933750
2	0.7899414800749480	0.9383044161905260	0.9150916335933750
2	0.9566081467416150	0.9383044161905260	0.9150916335933750
/



Also, it there a way to run alm in parallel?
Here is my job file:
#!/bin/bash
#SBATCH --job-name="Alamode-InS"
#SBATCH -o qe.%j.out
#SBATCH -e qe.%j.err
### Change partition if needed
#SBATCH --partition=large-shared
### Change the following to your allocation ID
#SBATCH -A alr105
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=128
#SBATCH --mem-per-cpu=10000M
#SBATCH --export=ALL
#SBATCH -t 48:00:00

### Set up environment
#module reset
#module load cpu/0.17.3b
#module load gcc/10.2.0/npcyll4
#module load openmpi/4.1.3/oq3qvsv
#module load quantum-espresso/7.0/zl3kj67
cd /home/eesha276/alamode/

. /cm/shared/apps/spack/0.17.3/cpu/b/opt/spack/linux-rocky8-zen/gcc-8.5.0/anaconda3-2021.05-q4munrgvh7qp4o7r3nzcdkbuph4z7375/etc/profile.d/conda.sh
conda activate alamode
~/alamode/alamode/_build/alm/alm ins_alm.in > [ins_alm.log2[](url)]([url]())

**Reply by ttadano** [reply]
Please rerun the calculation using a computer node that has more RAM. Also, please ensure that the zero displacements in DFSET are numerically zero, e.g., 0.0000000. If these values deviate from zero (0.001...), the code assumes them to be nonzero and allocates the memory space for them even when the sparse solver is used.
